tracer/src/Datadog.Trace.Tools.Runner/RunCiCommand.cs

```diff
@@ -218,36 +218,50 @@ private async Task ExecuteAsync(InvocationContext context)
             // Let's set the code coverage datacollector if the code coverage is enabled
             if (codeCoverageEnabled)
             {
+                var isDotnetCommand = string.Equals(program, "dotnet", StringComparison.OrdinalIgnoreCase) ||
+                                      string.Equals(program, "dotnet.exe", StringComparison.OrdinalIgnoreCase);
+                var isVsTestCommand = string.Equals(program, "VSTest.Console", StringComparison.OrdinalIgnoreCase) ||
+                                      string.Equals(program, "VSTest.Console.exe", StringComparison.OrdinalIgnoreCase);
+
                 // Check if we are running dotnet process
-                if (string.Equals(program, "dotnet", StringComparison.OrdinalIgnoreCase) ||
-                    string.Equals(program, "VSTest.Console", StringComparison.OrdinalIgnoreCase))
+                if (isDotnetCommand || isVsTestCommand)
```

Comment:
Will we ever have `isDotnetCommand == true` _and_ `isVsTestCommand == true`? Is `VSTest.Console test` a thing you're looking for? If not, then I assume you don't need `|| isVsTestCommand` here? 🤔

Reply:
I need that `||` because the code to attach the collector is inside that condition.
We need to attach the collector, and there's two mode of attach:

1) `dotnet test` : Mode 1
2) `dotnet vstest` : Mode 2
3) `vstest.console` : Mode 2

So when I detect `dotnet` I need to look further for `test` or `vstest` to select the mode.

That's why initially I have `if (isDotnetCommand || isVsTestCommand)` (we need to attach the collector if we can)  and later the `if (isDotnetCommand)` to look into the arguments (to decide if we have test or vstest to attach to).

Reply:
If the answer to 1 and 2 about are both "Mode 2", then I think we should skip the arg checking if `isVsTestCommand == true` 🤔

Reply:
done in [f287b66](https://github.com/DataDog/dd-trace-dotnet/pull/5412/commits/f287b66011520eadecfbec7eb223fa2ada24f48c)